OISF/suricata
# Pcre2 3194 v5

Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket:
https://redmine.openinfosecfoundation.org/issues/3194

Describe changes:
- Use PCRE2 instead of pcre for transform pcrexform
- qa : remove some obsolete files
- pcrexform : checks that we have exactly one substring capture in a signature
- transforms : use generic `Free` functions everywhere
- migrate detect-pcre to pcre2
- 
After that, the goal is to remove pcre from all keywords parsing...

suricata-verify-pr: 480

https://github.com/OISF/suricata-verify/pull/480

Replaces https://github.com/OISF/suricata/pull/5991 with migrating detect-pcre to pcre2 (only for detection, not pcre keyword parsing itself...)
